### PR TITLE
Feature/modify vault for allocation adjustment

### PR DESCRIPTION
**Pull request by baizetianxia**
1 update exchange from exchangeAggregator to exchangeHelper
2 add targetDebt into StrategyParam and relative method
3 add 'totalValueInVaultBuffer' and 'valueOfTrackedTokensInVaultBuffer' method

### CODE REVIEW

File: contracts/exchanges/ExchangeHelper.sol [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        uint256 _fromAmount,
+        bytes calldata _calldata
+    ) internal returns (
+            bool _success, 
+            uint256 _returnAmount
+    ) {
+        bytes memory _result;
+
+        uint256 beforeBalOfToToken = _balanceOfToken(_toToken, address(this));
+        if (_fromToken == address(0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)) {
```
Comment (ZhangBrain) [inline contracts/exchanges/ExchangeHelper.sol:117]: use NativeToken.NATIVE_TOKEN replace 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE

File: contracts/exchanges/ExchangeHelper.sol [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
```
Comment (ZhangBrain) [inline contracts/exchanges/ExchangeHelper.sol:84]: use NativeToken.NATIVE_TOKEN replace 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE

File: contracts/exchanges/ExchangeHelper.sol [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+            IERC20Upgradeable(_fromToken).safeApprove(_oneInchRouter, _fromAmount);
+            (_success, _result) = _oneInchRouter.call(_calldata);
+
+        }
+        uint256 afterBalOfToToken = _balanceOfToken(_toToken, address(this));
+
+        if (!_success) {
+            revert(RevertReasonParser.parse(_result, "1inch V4 swap failed: "));
+        } else {
+            _returnAmount = afterBalOfToToken - beforeBalOfToToken;
```
Comment (ZhangBrain) [inline contracts/exchanges/ExchangeHelper.sol:97]: afterBalOfToToken can place in else logic.

File: contracts/vault/Vault.sol [hunk 4]
```diff
@@ -46,9 +50,15 @@ contract Vault is VaultStorage {
...
             minCheckedStrategyTotalDebt = 1e17;
+            minStrategyTargetDebt = 1e18;
         } else {
             minCheckedStrategyTotalDebt = 1000e18;
+            minStrategyTargetDebt = 2000e18;
         }
+
+        oneInchRouter = 0x1111111254EEB25477B68fb85Ed929f73A960582;
+        paraRouter = 0xDEF171Fe48CF0115B1d80b88dc8eAB59176FEe57;
+        paraTransferProxy = 0x216B4B4Ba9F3e719726886d34a177484278Bfcae;
```
Comment (ZhangBrain) [inline contracts/vault/Vault.sol:61]: Put it in the constructor of ExchangeHelper.

File: contracts/vault/VaultBuffer.sol [hunk 5]
```diff
@@ -82,6 +93,11 @@ contract VaultBuffer is
         _initAccessControl(_accessControlProxy);
 
         mDistributeLimit = 50;
+
+        valueInterpreter = _valueInterpreter;
+        oneInchRouter = 0x1111111254EEB25477B68fb85Ed929f73A960582;
+        paraRouter = 0xDEF171Fe48CF0115B1d80b88dc8eAB59176FEe57;
+        paraTransferProxy = 0x216B4B4Ba9F3e719726886d34a177484278Bfcae;
```
Comment (ZhangBrain) [inline contracts/vault/VaultBuffer.sol:95]: Put it in the constructor of ExchangeHelper.

File: contracts/vault/VaultStorage.sol [hunk 6]
```diff
@@ -302,6 +293,9 @@ contract VaultStorage is Initializable, ReentrancyGuardUpgradeable, AccessContro
     bytes32 internal constant ADMIN_IMPL_POSITION =
     0x3d78d3961e16fde088e2e26c1cfa163f5f8bb870709088dd68c87eb4091137e2;
 
+    /// @notice Minimum strategy target debt that will be checked when set strategy target debt
+    uint256 public minStrategyTargetDebt;
```
Comment (ZhangBrain) [inline contracts/vault/VaultStorage.sol:297]: need set funcation.

**Review by mingshengke — APPROVED** [review]

**Pull request merged**